Base VIPER Interfaces/LazyHostingViewController.swift

```diff
@@ -0,0 +1,52 @@
+//
+//  LazyHostingViewController.swift
+//  EmergencyApp
+//
+//  Created by Goran Brlas on 24.08.2022..
+//  Copyright © 2022 Infinum. All rights reserved.
+//
+
+import SwiftUI
+import UIKit
+
+/// Hosting controller that should be used in modules that use SwiftUI for
+/// creating their views. We are lazily adding the root view in order to
+/// conform to our VIPER module structure.
+///
+/// The issue is that we need to
+/// pass the presenter to our SwiftUI view as an ObservedObject, but to create
+/// the presenter we need to first create the wireframe (and VC), leading to a
+/// cycle that can only be broken by this lazy loading "hack".
+///
+/// Each `LazyHostingViewController` will be generic over the root view that is
+/// contained inside it.
+class LazyHostingViewController<RootView: View>: UIViewController {
+
+    var rootView: RootView!
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        // Create the hosting controller that will hold the SwiftUI view
+        // and add it as a child view controller to self in order to properly
+        // propagate all VC lifecycle events.
+        //
+        // You can see more about child VC handling here:
+        // https://developer.apple.com/documentation/uikit/view_controllers/creating_a_custom_container_view_controller
+        let hostingController = UIHostingController(rootView: rootView)
```

Comment:
Did you try playing out with `loadView` method and injecting this as a root view for `LazyHostingViewController` - with your current approach you always have this one view which will just be container which I'm not fond of. It could be it won't work but give it a try please. Something like this: https://swiftrocks.com/writing-cleaner-view-code-by-overriding-loadview

Reply:
Why didn't you chose the path to create different kind of an wireframe for SwiftUI? Seems we are fighting with UIKit with this kind of an approach :/ 

I'm thinking about the cases where we'll have tens of child view controllers - every one of them will have that empty view behind it

Reply:
> Did you try playing out with loadView method and injecting this as a root view for LazyHostingViewController

I haven't at the time of writing, but I did try it out now. 

UIHostingController is a UIViewController, so it can't be set as the root view, and doing something like `view = UIHostingController(rootView: rootView).view` also doesn't work because of `UIViewControllerHierarchyInconsistency` - `A view can only be associated with at most one view controller at a time!` which makes sense.

> I'm thinking about the cases where we'll have tens of child view controllers - every one of them will have that empty view behind it

We'll cover this as two separate cases: 
* main screen module -> this one will use the templates seen here
* child SwiftUI modules that should be embedded in the main one -> this one will use a different generator that will create SwiftUI-only modules (basically, wireframes will be SwiftUI views as well)

Doing things like this will avoid this issue of x empty views, there will always be only one per screen. However, this second case still needs to be brainstormed a bit, but we'll probably go with something @Truba has already done in another project.